### PR TITLE
Wildhammer Facial Tattoos

### PR DESCRIPTION
**Pull request by ercarp**
## Changelog:
- Added **8** facial tattoos for Wildhammer Dwarves:
  - `facial_wildhammer_tattoo_01` ("Aerie")
  - `facial_wildhammer_tattoo_02` ("Hinterlands")
  - `facial_wildhammer_tattoo_03` ("Sky"), **disabled for female dwarves**,
  - `facial_wildhammer_tattoo_04` ("Thunder") **disabled for female dwarves**,
  - `facial_wildhammer_tattoo_05` ("Highlands")
  - `facial_wildhammer_tattoo_05_f` ("Northeron")
  - `facial_wildhammer_tattoo_06` ("Gryphon") **disabled for female dwarves**,
  - `facial_wildhammer_tattoo_06_f` ("Batol")
- Added localization to all of the markings. 
- Gave **Kurdran** the "Thunder" marking and **Falstad** the "Gryphon" marking.
  - Also disabled headgear for both characters.

## Images:

<details><summary>Click to expand</summary>

**Aerie** (`facial_wildhammer_tattoo_01`)
image
image

**Hinterlands** (`facial_wildhammer_tattoo_02`)
image
image

**Sky** (`facial_wildhammer_tattoo_03`)
image
image

**Thunder** (`facial_wildhammer_tattoo_04`)
image
image

**Highlands** (`facial_wildhammer_tattoo_05`)
image
image

**Northeron** (`facial_wildhammer_tattoo_05_f`)
image
image

**Gryphon** (`facial_wildhammer_tattoo_06`)
image
image

**Batol** (`facial_wildhammer_tattoo_06_f`)
image
image

</details>

## Known issues:
- ~~Characters generated with `facial_wildhammer_tattoo_03` or `facial_wildhammer_tattoo_04` should be bald for their tattoos to show fully.~~
  - Fixed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/commit/9e2f0aa997b13100e94cb0468d3bd8a11b73eeb3.
- Some markings will appear to have a line going through the top middle of the character's head. This is an issue with CK3, not an issue with the markings. If you zoom in on the character, the line disappears.

<details><summary>Click to expand</summary>

image
image

</details>

## Tasks:
- [x] Add the remaining male facial tattoos:
  - [x] Add `facial_wildhammer_tattoo_05` ("Highlands").
  - [x] Add `facial_wildhammer_tattoo_06` ("Gryphon").
  - [ ] _(Optional)_ Add `facial_wildhammer_tattoo_07` ("Ironforge").
    - Based on the name, maybe this was intended to be a Bronzebeard tattoo, but we could still use it for Wildhammer Dwarves.
- [x] Add female facial tattoos:
  - [x] Add `facial_wildhammer_tattoo_05_f` ("Highlands").
  - [x] Add `facial_wildhammer_tattoo_06_f` ("Gryphon").
- [x] Localize the tattoos.
- [x] Wildhammer Dwarves with prominent head tattoos should be made bald (see **Tasks**).

## Tests:
- [x] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [x] The mod takes less than 5.5 GB in the Task Manager (Windows)

# How to test:

There are a few ways to test these markings:

**A**. Open up the game in [Debug Mode](https://ck3.paradoxwikis.com/Console_commands#Enabling_debug_mode) and view them in the **Portrait Editor** under `gene_facial_markings`.
**B**. View the markings in **Ruler Designer**.
**C**. Start a game and look at characters with the **_Wildhammer_** culture to see how the markings look in-game. **Some dwarves have no markings, this is intended and normal.**

Provide feedback on the look of the markings and the localization used for their names (all names except Batol and Northeron are taken directly from the names they used for the markings in WoW).

### CODE REVIEW

File: common/culture/cultures/dwarf.txt [hunk 1]
```diff
@@ -404,6 +404,7 @@
 	wildhammer = {
```
Comment (zumbak04) [inline common/culture/cultures/dwarf.txt:404]: Will you add markings to Falstad and Kurdran?

Comment (ercarp) [reply common/culture/cultures/dwarf.txt:404]: I think I would prefer to do that on `dna-5`.

**Comment by ercarp** [comment]
@zumbak04 Since you asked about the markings for Kurdran and Falstad, I decided to look deeper into what kind of markings they actually use.

### Kurdran Wildhammer

<details><summary>Click to expand</summary>

**Art**
In artwork, Kurdran is most often depicted with the "Thunder" marking.
image
image

This one is a bit different, but it looks very similar.
image

**In-Game**
However, his character in-game uses markings that are basically identical to the "Gryphon" markings available to players (which I have yet to add, but I'm planning to).

image
image

</details>

### Falstad Wildhammer

<details><summary>Click to expand</summary>

**Art**
This is artwork for Heroes of the Storm, where he seems to have custom markings.
image

"Thunder" marking.
image

"Hinterlands" marking.
image

"Gryphon" marking.
image

**Heroes of the Storm**
Strangely, while it's similar to the art used for his Heroes of the Storm character, it's not the exact same. At least not the markings on his body.
image

**In-Game**
In WoW, his character model uses the same "Gryphon" markings as Kurdran.
image

</details>

So basically, with Kurdran we can 100% get away with giving him either the "Thunder" or "Gryphon" markings.

With Falstad, we *could* try to recreate his Heroes of the Storm markings since that's his most recent canon appearance, but he has also been depicted with other markings. And his WoW character is still using the same "Gryphon" markings as Kurdran.

**Review by CantTouchThisJinTao — APPROVED** [review]
All of the markings look good, definitely something that'll make playing a wildhammer dwarf appealing.
Not too sure about names as I can't remember much about them from WoW and can't find a lot of info on them atm.

**Comment by zumbak04** [comment]
> So basically, with Kurdran we can 100% get away with giving him either the "Thunder" or "Gryphon" markings.

Yes, please.

> With Falstad, we _could_ try to recreate his Heroes of the Storm markings since that's his most recent canon appearance, but he has also been depicted with other markings. And his WoW character is still using the same "Gryphon" markings as Kurdran.

I'm fine with both.

**Review by zumbak04 — APPROVED** [review]
LGTM! Good job!

**Review by Darknight367 — APPROVED** [review]
Looks neat!

**Review by Remerod — CHANGES_REQUESTED** [review]
First of, I approve of the markings, they look fantastic!
The only problem i would have is that some looks weird with hair,
So i would like to request if it is possible to have the markings that covers the head/scalp on bald or balding hairstyles only. I'll show some examples;

<details>
<summary>Click to expand</summary>

image
This looks ok

image
This looks weird Highlands

image
like this would definitely look better without hair


    facial_wildhammer_tattoo_01 ("Aerie") ok with hair
    facial_wildhammer_tattoo_02 ("Hinterlands") kinda ok with hair
    facial_wildhammer_tattoo_03 ("Sky"), disabled for female dwarves, work better without hair
    facial_wildhammer_tattoo_04 ("Thunder") disabled for female dwarves, 
    facial_wildhammer_tattoo_05 ("Highlands") don't work well with full hair in my opinion, bald or balding hairstyles work better with this in my opinion
    facial_wildhammer_tattoo_05_f ("Northeron") works with hair
    facial_wildhammer_tattoo_06 ("Gryphon") disabled for female dwarves, Don't work with hair
    facial_wildhammer_tattoo_06_f ("Batol") I think this looks better bald but workd with hair
    
image



I think you've done a fantastic job researching and making this tattoos/markings and i understand that the female variations would probably take the hardest hit by this. To make the absolute best of this, i would advise adding some more for females. But this is not necessary, my current concern and reason i requested this change is primarily (facial_wildhammer_tattoo_05) on females with hair, it looks weird. but that's my opinion. And i can bend on that.
As i said, this most basicly really good, and the only problem i have is with some hair combos vs bald.

I googled [warcraft wildhammer female dwarf] for some tattoos suggestions if you wanna add some for females, but this is just a small suggestion and not the reason i requested changes
image

image

image

image

image

image

And just to give you a reason to tell me that i am wrong, here is one of the (facial_wildhammer_tattoo_05) on female in game.
image


So just tell me if anything i said made some point, otherwise i'll go right ahead and approve this!

</details>

**Comment by ercarp** [comment]
> The only problem i would have is that some looks weird with hair,
> So i would like to request if it is possible to have the markings that covers the head/scalp on bald or balding hairstyles only. I'll show some examples;

All of the markings that go directly on top of a character's head have already been disabled from characters with hair. Kurdran and Falstad are the only exceptions to this as they have custom DNA. Markings that go only slightly under hair, but not directly on top of their heads, are fine. If you go in-game and look at some Wildhammer dwarves, you will see this in effect.

<details><summary>Click to expand</summary>

image
image
image

</details>

> `facial_wildhammer_tattoo_03` ("Sky"), disabled for female dwarves, work better without hair
> `facial_wildhammer_tattoo_06` ("Gryphon") disabled for female dwarves, Don't work with hair

These markings are already disabled for characters with hair.

> `facial_wildhammer_tattoo_05` ("Highlands") don't work well with full hair in my opinion, bald or balding hairstyles work better with this in my opinion

I have to disagree as the main focus of the tattoo is not on the top of the head. Most of the tattoo can be seen directly from the front, and it's not obscured by the hair to any significant degree. It can also be used by characters with hair in WoW and it looks fine to me.

<details><summary>Click to expand</summary>

image
image

</details>

> `facial_wildhammer_tattoo_06_f` ("Batol") I think this looks better bald but workd with hair

This was primarily intended to be used by characters with hair, since it is a tattoo used by female dwarves in WoW, but it was still necessary to create the full tattoo underneath so it doesn't look strange on bald characters.

<details><summary>Click to expand</summary>

image
image

</details>

> I googled [warcraft wildhammer female dwarf] for some tattoos suggestions if you wanna add some for females, but this is just a small suggestion and not the reason i requested changes

All of these look wonderful, but I am a bit hesitant about using markings from fan art and would like to stick to only using the official ones from WoW for the time being.

It's also worth considering that the Wildhammer markings currently take up 8 decal slots, which might not seem like much, but when you consider how many other races need some extra marking flavor added, it does add up. I think we should try to keep the number of markings for each culture at around 6-8 max (the blood trolls are the exception to this, as I made them before we knew about the 100 decal limit).

**Review by Remerod — APPROVED** [review]
@ercarp I hear your arguments. And as i said, it was minor things that i pointed out that was very opinion heavy on my part. So i see no reason not to approve ;)
